addons/calendar/models/calendar_event.py

```diff
@@ -1004,32 +1010,107 @@ def _rewrite_recurrence(self, values, time_values, recurrence_values):
                 stop = base_time_values['stop'] + (stop_update - self.stop)
                 stop_date = base_time_values['stop'].date() + (stop_update.date() - self.stop.date())
                 update_dict.update({'stop': stop, 'stop_date': stop_date})
+        return update_dict
+
+    @api.model
+    def _get_archive_values(self):
+        """ Return parameters for archiving events in calendar module. """
+        return {'active': False}
+
+    @api.model
+    def _check_values_to_sync(self, values):
+        """ Method to be overriden: return candidate values to be synced within rewrite_recurrence function scope. """
+        return False
+
+    @api.model
+    def _get_update_future_events_values(self):
+        """ Return parameters for updating future events within _update_future_events function scope. """
+        return {}
+
+    @api.model
+    def _get_remove_sync_id_values(self):
+        """ Return parameters for removing event synchronization id within _update_future_events function scope. """
+        return {}
 
+    @api.model
+    def _get_updated_recurrence_values(self, new_start_date):
+        """ Copy values from current recurrence and update the start date weekday. """
+        [previous_recurrence_values] = self.recurrence_id.copy_data()
+        if self.start.weekday() != new_start_date.weekday():
+            previous_recurrence_values.pop(weekday_to_field(self.start.weekday()), None)
+        return previous_recurrence_values
+
+    @api.model
```

Comment:
This method relies on the records in self, so it should not be decorated with api.model